src/adapter/src/coord/introspection.rs

```diff
@@ -134,6 +134,11 @@ pub fn auto_run_on_introspection<'a, 's, 'p>(
         | Plan::SideEffectingFunc(_) => return TargetCluster::Active,
     };
 
+    // Use transaction cluster if we're mid-transaction
+    if let Some(cluster_id) = session.transaction().cluster() {
```

Comment:
For my other comment: I think this is why it's ok (maybe better) to use an assert instead of an error, because the add_ops fn should never get a conflicting cluster id.